src/button-group/menu-dropdown-item.tsx

```diff
@@ -0,0 +1,84 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useRef, useState } from 'react';
+import { ButtonGroupProps } from './interfaces';
+import { ButtonDropdownProps } from '../button-dropdown/interfaces';
+import { CancelableEventHandler, fireCancelableEvent } from '../internal/events';
+import ButtonDropdown from '../button-dropdown/internal';
+import Tooltip from './tooltip';
+import styles from './styles.css.js';
+
+const MenuDropdownItem = React.forwardRef(
+  (
+    {
+      item,
+      onItemClick,
+      dropdownExpandToViewport,
+    }: {
+      item: ButtonGroupProps.MenuDropdown;
+      onItemClick?: CancelableEventHandler<ButtonGroupProps.ItemClickDetails>;
+      dropdownExpandToViewport?: boolean;
+    },
+    ref: React.Ref<ButtonDropdownProps.Ref>
+  ) => {
+    const containerRef = useRef<HTMLDivElement>(null);
+    const [popoverOpen, setPopoverOpen] = useState(false);
+    const [isActionPopover, setIsActionPopover] = useState(false);
```

Comment:
The popover code is similar between button and menu dropdown, maybe we can move it to the item-element or reuse in some other way?